src/redfish/discovery/discovery.py

```diff
@@ -72,6 +72,8 @@ def discover_ssdp(port=1900, ttl=2, response_time=3):
     sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM, socket.IPPROTO_UDP)
     sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
     sock.setsockopt(socket.IPPROTO_IP, socket.IP_MULTICAST_TTL, ttl)
+    if iface:
+        sock.setsockopt(socket.SOL_SOCKET, socket.SO_BINDTODEVICE, iface+'\0')

```

Comment:
We may have to pass `bytes(iface + "\0")` to be compatible with python3

Reply:
@MoshiBin not sure you are right...
```
[root@localhost ~]# python2 -c "print bytes('eth0'+'\0')"
eth0
```
vs
```
[root@localhost ~]# python3 -c "print bytes('eth0'+'\0')"
  File "<string>", line 1
    print bytes('eth0'+'\0')
              ^
SyntaxError: invalid syntax
```